scipy/stats/_morestats.py

```diff
@@ -987,7 +989,8 @@ def boxcox_llf(lmb, data, *, axis=0, keepdims=False, nan_policy='propagate'):
                           result_to_tuple=lambda x: (x,))
 def _boxcox_llf(data, axis=0, *, lmb):
     xp = array_namespace(data)
-    lmb, data = xp_promote(lmb, data, force_floating=True, xp=xp)
+    dtype = xp_result_type(lmb, data, force_floating=True, xp=xp)
+    data = xp.asarray(data, dtype=dtype)
```

Comment:
It would be nice if `xp_promote` could convert Python scalar `lmb` to an array on the right device, but `xp_promote` uses our `_asarray`, which doesn't support `device`. I'd like to leave it as-is in this PR, but it's something to keep in mind in gh-22049.